### PR TITLE
Added files discovery method

### PR DESCRIPTION
**Pull request by sabaubogdan**

### CODE REVIEW

**Comment by sabaubogdan** [comment]
.

File: src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java [hunk 1]
```diff
@@ -2,13 +2,58 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
 
 @Service
 public class FileDiscoveryServiceImpl implements FileDiscoveryService {
 
+    public static void main(String[] args) {
```
Comment (sabaudragos) [inline src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java]: The `main()` method should be removed. It's used only for testing purposes

File: src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java [hunk 2]
```diff
@@ -2,13 +2,58 @@
...
+                        docList.add(filePath.toString());
+                    } else if (filePath.toString().endsWith("pdf")) {
+                        pdfList.add(filePath.toString());
+                    } else {
+                        unknownFormat.add(filePath.toString());
+                    }
+                }
+            });
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
```
Comment (sabaudragos) [inline src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java]: We need a logger here. Please add one on the class `@Slf4j` and then log here a message like `log.error('Error while discovering files', e)`

File: src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java [hunk 3]
```diff
@@ -2,13 +2,58 @@
...
+        }
+
+        Map<String, List<String>> foundFiles = new HashMap<>();
```
Comment (sabaudragos) [inline src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java]: A more appropriate name would be `filePathsFound`

File: src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java [hunk 4]
```diff
@@ -2,13 +2,58 @@
...
+
+        List<String> txtList = new ArrayList();
+        List<String> docList = new ArrayList();
+        List<String> pdfList = new ArrayList();
+        List<String> unknownFormat = new ArrayList();
+
+        try (Stream<Path> paths = Files.walk(Paths.get(directoryPath), Integer.MAX_VALUE)) {
+            paths.forEach(filePath -> {
+                if (Files.isRegularFile(filePath)) {
+                    if (filePath.toString().endsWith("txt")) {
```
Comment (sabaudragos) [inline src/main/java/xyz/vegaone/scribe/service/FileDiscoveryServiceImpl.java]: A good practice is to not use text directly into code. A better approach is to define constants at the class level and then use those constants int the code. 
e.g. 
```java
private static final String TEXT_FILE_FORMAT = "txt";
private static final String PDF_FILE_FORMAT = "pdf";
```

**Review by sabaudragos — CHANGES_REQUESTED** [review]

**Review by sabaudragos — APPROVED** [review]

**Pull request merged**